### ISSUE TITLE
Filter swear words with asterisk

### ISSUE THREAD

**Issue by ZGGrOydHRyl** [FrostCo/AdvancedProfanityFilter#580]
#### :bug: Description

AdvancedProfanityFilter is not filtering swear words with asterisk unless I manually add them to the list.  I can't add all swear words with asterisk manually.  AdvancedProfanityFilter should filter it or provide a setting to filter it.

#### :twisted_rightwards_arrows: Steps To Reproduce

 Swear words with asterisk (example: "t*sting") isn't being filtered. Is there a setting I can enable to filter all swear words with asterisk?

Screenshot of Reddit post:
![image](https://github.com/user-attachments/assets/1e98df8b-6ac3-4b9a-a577-30d6f3b6827b)

#### :heavy_check_mark: Expected behavior

swear words with asterisk should be filtered.

#### :camera: Screenshots
Here  is a example from reddit post where AdvancedProfanityFilter  isn't filtering swear words.

![image](https://github.com/user-attachments/assets/1e98df8b-6ac3-4b9a-a577-30d6f3b6827b)

#### :computer: Details

- **Device:** [PC]
- **OS:** [Windows 10]
- **Browser:** [Chrome]
- **Browser Version:** [129.0]
- **Affected site(s):** [www.reddit.com]

**Comment by richardfrost** [comment]
@ZGGrOydHRyl Thanks for the suggestion. I'll look into this more, but my first thought is this will be difficult to accomplish in a reliable way. Currently the filter does support matching separators (`-` & ` _`) in a word, like `t-e-s-t-i-n-g` or `test-ing`, but still requires all the letters to be there so we know it is the word we are looking for, just with characters to separate it (commonly used in an attempt to bypass filtering on a site).

**Comment by ZGGrOydHRyl** [comment]
@richardfrost For a quick resolution or a workaround can you not just import a list of swear words with asterisks? Or let users import it through extension? I'm sure we can find a list of swear words with asterisk online.

Allowing us to import a list would be nice. I can't type all of it but I can find a list and import it if the extension allows.

**Comment by richardfrost** [comment]
@ZGGrOydHRyl Yes, that is something you could do. When adding a lot (100+ words) it can cause the filter to slow down on slower devices, but most shouldn't have much of an issue.

To import a list of words, open the Options page and go to the "Words" tab, and click the `Bulk Editor` button. In the bottom-left corner you can paste a list (one word/phrase per line) and then click the `Bulk Add Words` button. That will then add them to the table above, where you can make any modifications if you'd like. Once it all looks good, just click the `Save` button.

Before using the bulk word editor I'd recommend saving a backup of your settings (Options page > Config tab > Export).

**Comment by ZGGrOydHRyl** [comment]
@richardfrost I prefer to filter all words which has asterisk in it and replace it with something else like "<word_filtered>". How can I achieve this?

**Comment by richardfrost** [comment]
@ZGGrOydHRyl, you should be able to accomplish that by adding a new word:
- Word/Phrase : `\w+[*]+(\w+|)`
- Substitution: `<word_filtered>`
- Match Method: `Regex`

![image](https://github.com/user-attachments/assets/26f2c7a2-bb9a-4b8f-8af0-4c83f05f2a5b)

![image](https://github.com/user-attachments/assets/b6da8dd2-e872-4067-bd1d-d82795d83045)

**Comment by ZGGrOydHRyl** [comment]
@richardfrost That worked thank you!

**Comment by richardfrost** [comment]
@ZGGrOydHRyl You're welcome! Let me know if you run into trouble with it, its possible it might need to be adjusted a bit, but hopefully it works well for you!
